server/router/driver.ts

```diff
@@ -12,6 +12,52 @@ import { UserType } from '../models/subscription';
 const router = express.Router();
 const tableName = 'Drivers';
 
+// Get all drivers
+router.get('/', validateUser('Admin'), (req, res) => {
+  db.getAll(res, Driver, tableName);
+});
+
+// Get all available drivers at a specific date and time
+router.get('/available', validateUser('User'), (req, res) => {
+  const { date, startTime: reqStartTime, endTime: reqEndTime } = req.query;
+  const reqDay = moment(date as string).day();
+
+  if ((reqStartTime as string) >= (reqEndTime as string)) {
+    res.status(400).send({ err: 'startTime must precede endTime' });
+  }
+
+  db.getAll(res, Driver, tableName, (doc: DriverType[]) => {
+    const drivers = doc.filter((driver) => {
+      const availStart = (() => {
+        if (reqDay === 2) return driver.availability.Tue?.startTime;
+        if (reqDay === 1) return driver.availability.Mon?.startTime;
+        if (reqDay === 3) return driver.availability.Wed?.startTime;
+        if (reqDay === 4) return driver.availability.Thu?.startTime;
+        if (reqDay === 5) return driver.availability.Fri?.startTime;
```

Comment:
i think your original approach of getting the day as a string (Mon, Tue, etc.), then finding `driver.availability[day].startTime` or `endTime` was better. doing it like this results in a lot of redundant code

Reply:
Done